### PR TITLE
Fix #1456 getting `nil` from account data has `"nil"` result

### LINKED ISSUE #1456: Getting nil from account data has "nil" result
**Describe the bug**

Using `setAccountData(acc, "key", nil)` will return `"nil"` instead of `nil` when it is getted.

**To reproduce**

1. `run setAccountData(getAccount("Console"), "test", nil) == true`
2. `run getAccountData(getAccount("Console", "test") == "nil"`

**Expected behaviour**

`run getAccountData(getAccount("Console", "test") == nil`

**Screenshots**

image

**Version**

1.5.7 36efacb36be9e6ba5d60151b26506c6f3f9ae75f

**Additional context**

The correct `iType` is being stored in the database, but it is not being checked for `LUA_TNIL`. https://github.com/multitheftauto/mtasa-blue/blob/36efacb36be9e6ba5d60151b26506c6f3f9ae75f/Server/mods/deathmatch/logic/CAccountManager.cpp#L685-L694

https://github.com/multitheftauto/mtasa-blue/blob/36efacb36be9e6ba5d60151b26506c6f3f9ae75f/Server/mods/deathmatch/logic/CAccountManager.cpp#L705-L739

### PR DESCRIPTION
**Pull request by Pirulax**
Fixes #1456  + small code refactor

### CODE REVIEW

File: Server/mods/deathmatch/logic/CAccountManager.cpp [hunk 1]
```diff
@@ -676,22 +676,32 @@ std::shared_ptr<CLuaArgument> CAccountManager::GetAccountData(CAccount* pAccount
...
+        case LUA_TNUMBER:
+            pResult->ReadNumber(strtod(value, NULL));
+            break;
+
+        case LUA_TNIL:
+            break;
+
+        case LUA_TSTRING:
+            pResult->ReadString(value);
+        default:
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CAccountManager.cpp:703]:
```suggestion

        default:
```

**Review by qaisjp — CHANGES_REQUESTED** [review]
This PR doesn't fix the issue entirely because the cache doesn't return `nil` either:

https://github.com/multitheftauto/mtasa-blue/blob/6c6a1248adcdd9715c542a7dbdd05a5a2f99ee0e/Server/mods/deathmatch/logic/CAccountManager.cpp#L656-L660

https://github.com/multitheftauto/mtasa-blue/blob/6c6a1248adcdd9715c542a7dbdd05a5a2f99ee0e/Server/mods/deathmatch/logic/CAccount.cpp#L115-L140

**Comment by Pirulax** [comment]
my bad..

**Comment by Pirulax** [comment]
There you go.

**Comment by Pirulax** [comment]
Il open a separate PR to refactor the code.
A few lines.

**Review by qaisjp — APPROVED** [review]
thanks!

**Pull request merged**